storage/client/src/main/java/com/azure/storage/common/credentials/SharedKeyCredential.java

```diff
@@ -0,0 +1,185 @@
+
+    // Pieces of the connection string that are needed.
+    private static final String ACCOUNT_NAME = "AccountName".toLowerCase();
+    private static final String ACCOUNT_KEY = "AccountKey".toLowerCase();
+
+    private final String accountName;
+    private final byte[] accountKey;
+
+    /**
+     * Initializes a new instance of SharedKeyCredentials contains an account's name and its primary or secondary
+     * accountKey.
+     *
+     * @param accountName The account name associated with the request.
+     * @param accountKey The account access key used to authenticate the request.
+     */
+    public SharedKeyCredential(String accountName, String accountKey) {
+        this.accountName = accountName;
+        this.accountKey = Base64.getDecoder().decode(accountKey);
+    }
+
+    /**
+     * Creates a SharedKey credential from the passed connection string.
+     * @param connectionString Connection string used to build the SharedKey credential.
+     * @return a SharedKey credential if the connection string contains AccountName and AccountKey
+     * @throws IllegalArgumentException If {@code connectionString} doesn't have AccountName or AccountKey.
+     */
+    public static SharedKeyCredential fromConnectionString(String connectionString) {
+        HashMap<String, String> connectionStringPieces = new HashMap<>();
+        for (String connectionStringPiece : connectionString.split(";")) {
+            String[] kvp = connectionStringPiece.split("=", 2);
+            connectionStringPieces.put(kvp[0].toLowerCase(), kvp[1]);
+        }
+
+        String accountName = connectionStringPieces.get(ACCOUNT_NAME);
+        String accountKey = connectionStringPieces.get(ACCOUNT_KEY);
+
+        if (ImplUtils.isNullOrEmpty(accountName) || ImplUtils.isNullOrEmpty(accountKey)) {
+            throw new IllegalArgumentException("Connection string must contain 'AccountName' and 'AccountKey'.");
+        }
+
+        return new SharedKeyCredential(accountName, accountKey);
+    }
+
+    /**
+     * Generates the SharedKey Authorization value from information in the request.
+     * @param requestURL URL of the request
+     * @param httpMethod HTTP method being used
+     * @param headers Headers on the request
+     * @return the SharedKey authorization value
+     */
+    public String generateAuthorizationHeader(URL requestURL, String httpMethod, Map<String, String> headers) {
+        return computeHMACSHA256(buildStringToSign(requestURL, httpMethod, headers));
+    }
+
+    private String buildStringToSign(URL requestURL, String httpMethod, Map<String, String> headers) {
+        String contentLength = headers.get("Content-Length");
+        contentLength = contentLength.equals("0") ? "" : contentLength;
+
+        // If the x-ms-header exists ignore the Date header
+        String dateHeader = (headers.containsKey("x-ms-date")) ? "" : headers.getOrDefault("Date", "");
```

Comment:
I have tested what happen if I put empty string for date. It did not pass at time.

At the time we tested the API, the input and output I got is like:
Header as input

```
headers: {
 Content-Length:0
 If-Match: null
 ...
}
```

The string after we built:
```
0
null

```
Null are not supposed to appear in the string. This is what I observed. 
Can we double check the SharedKey in the Tests we build?